File: src/main/java/tk/hiddenname/smarthome/entity/Device.java
package tk.hiddenname.smarthome.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "device")
@EqualsAndHashCode(of = {"id"})
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Device {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", updatable = false, nullable = false)
    private Integer id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private Boolean reverse;

    @Column(updatable = false, name = "creation_date")
    @JsonFormat(shape = JsonFormat.Shape.OBJECT, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime creationDate;

    @JoinColumn(name = "gpio_id", nullable = false, updatable = false)
    @OneToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private GPIO gpio;
}

File: src/main/java/tk/hiddenname/smarthome/model/hardware/Device.kt
package tk.hiddenname.smarthome.model.hardware

import com.fasterxml.jackson.databind.PropertyNamingStrategy.SnakeCaseStrategy
import com.fasterxml.jackson.databind.annotation.JsonNaming
import javax.persistence.Entity
import javax.persistence.Table

@Entity
@Table(name = "device")
@JsonNaming(SnakeCaseStrategy::class)
class Device : Hardware()